File: backend/amigos_dotnet_backend/Models/Entities/UserSettings.cs
using SocialMediaAdminBackend.Models.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SocialMediaAdminBackend.Models.Entities;

[Table("user_settings")]
public class UserSettings
{
    [Key]
    [Column("user_id")]
    public long UserId { get; set; }

    [Column("email_notifications")]
    public bool EmailNotifications { get; set; } = true;

    [Column("message_notifications")]
    public bool MessageNotifications { get; set; } = true;

    [Column("push_notifications")]
    public bool PushNotifications { get; set; } = true;

    [Column("account_visibility")]
    public string AccountVisibility { get; set; } = "PUBLIC";

    [Column("language")]
    public string Language { get; set; } = "en";

    [Column("theme")]
    public string Theme { get; set; } = "LIGHT";

    [ForeignKey("UserId")]
    public User? User { get; set; }
}

File: backend/amigos_java_backend/src/main/java/com/socialmedia/entity/UserSettings.java
package com.socialmedia.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "user_settings")
@Getter
@Setter
public class UserSettings {

    @Id
    private Long userId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "email_notifications")
    private boolean emailNotifications = true;

    @Column(name = "message_notifications")
    private boolean messageNotifications = true;

    @Column(name = "push_notifications")
    private boolean pushNotifications = true;

    @Column(name = "account_visibility")
    private String accountVisibility = "PUBLIC"; // PUBLIC, PRIVATE

    @Column(name = "language")
    private String language = "en";

    @Column(name = "theme")
    private String theme = "LIGHT"; // LIGHT, DARK, SYSTEM
}